crypto/src/asset.rs

```diff
@@ -89,6 +96,31 @@ impl Id {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct DenomUnit {
+    /// The name of this denomination, e.g. `upenumbra`.
+    pub denom: String,
+    /// The exponent of this denomination, if the minimum denomination, `0`.
+    pub exponent: u8,
+    /// A list of alternative aliases for the denomination.
+    pub aliases: Vec<String>,
+}
+
+/// Metadata about each asset including the minimum denomination.
+///
+/// Based on [ADR-024](https://docs.cosmos.network/master/architecture/adr-024-coin-metadata.html).
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+pub struct Metadata {
+    /// The description of this asset, e.g. `The native token for the Penumbra zone.`
+    pub description: String,
+    /// A list of alternative denominations that can be used, e.g. `upenumbra`, `penumbra`.
+    pub denom_units: Vec<DenomUnit>,
```

Comment:
For validator staking token assets we'll probably only have a single denomination, right?

Reply:
For validators I think we'll want to mirror whatever we decide on for `penumbra`, e.g. so I could imagine there eventually being a `udPEN-{validatorkey}` if we have `upenumbra` etc. In terms of how to define the metadata for all the delegation assets, I think we'll want some template metadata that we can reuse for every delegation token, I'll file an issue about this